src/citrine/jobs/waiting.py

```diff
@@ -0,0 +1,133 @@
+    status_text = status.status
+    output_text = status_text if status_text != "InProgress" else "In progress"
+    return output_text
+
+
+def _print_execution_status(
+    status: WorkflowExecutionStatus, start_time: float, line_end: str = "\r"
+):
+    print(
+        "Status = {:<25}Elapsed time".format(_pretty_execution_status(status)),
+        " = {}s".format(str(int(time() - start_time)).rjust(3)),
+        end=line_end,
+    )
+
+
+def wait_while_validating(
+    collection: Collection[Module],
+    module: Module,
+    print_status_info: bool = False,
+    timeout: float = 1800.0,
+    interval: float = 3.0,
+):
+    """
+    Wait until module is validated.
+
+    Parameters
+    ----------
+    collection : Collection[Module]
+        Collection containing module
+    module : Module
+        Module in Collection
+    print_status_info : bool, optional
+        Whether to print status info, by default False
+    timeout : float, optional
+        Maximum time spent inquiring in seconds, by default 1800.0
+    interval : float, optional
+        Inquiry interval in seconds, by default 3.0
+
+    Returns
+    -------
+    Module
+        Module in Collection
+
+    Raises
+    ------
+    ConditionTimeoutError
+        If fails to validate within timeout
+
+    """
+    def is_validated():
+        status = collection.get(module.uid).status
+        _print_validation_status(status, start)
+        return status != "VALIDATING"
+
+    start = time()
+
+    while not is_validated() and time() - start < timeout:
+        sleep(interval)
+    if not is_validated():
+        msg = "Timeout reached, but condition is still {}".format(is_validated())
+        raise ConditionTimeoutError(msg)
+
+    if print_status_info:
+        print("\nStatus info:")
+        status_info = collection.get(module.uid).status_info
+        pprint(status_info)
+
+    return collection.get(module.uid)
+
+
+def wait_while_executing(
+    execution: WorkflowExecution,
+    timeout: float = 1800.0,
+    interval: float = 3.0,
+):
+    """
+    Wait until execution is finished.
+
+    Parameters
+    ----------
+    execution : WorkflowExecution
+        WorklflowExecution to monitor
+    timeout : float, optional
+        Maximum time spent inquiring in seconds, by default 1800.0
+    interval : float, optional
+        Inquiry interval in seconds, by default 3.0
+
+
+    Raises
+    ------
+    ConditionTimeoutError
+        If fails to finish execution within timeout
+
+    """
+    def execution_is_finished():
+        status = execution.status()
+        _print_execution_status(status, start)
+        return not execution.status().in_progress
+
+    start = time()
+
+    while not execution_is_finished() and (time() - start < timeout):
+        sleep(interval)
+    if not execution_is_finished():
+        msg = "Timeout reached, but condition is still {}".format(execution_is_finished())
+        raise ConditionTimeoutError(msg)
```

Comment:
```suggestion
        raise ConditionTimeoutError(msg)
    return execution
```
See comment on the signature.